middleware/retry.go

```diff
@@ -20,13 +20,19 @@ func isPermanentFailure(err error) bool {
 	return !(errors.Is(err, psrpc.ErrNoResponse) || errors.Is(err, psrpc.ErrRequestTimedOut))
 }
 
-func retry(opt RetryOptions, fn func(timeout time.Duration) error) error {
+func retry(opt RetryOptions, done <-chan struct{}, fn func(timeout time.Duration) error) error {
 	attempt := 1
 	timeout := opt.Timeout
 	for {
-		if err := fn(timeout); err == nil || isPermanentFailure(err) || attempt == opt.MaxAttempts {
+		err := fn(timeout)
```

Comment:
should we change fn's signature to pass the context as well?

Reply:
no. retry doesn't use context it uses the `ctx.Done()` channel to determine whether the otherwise-retriable `ErrRequestTimedOut` was propagated from the parent context or the wrapped context we created in the request code here https://github.com/livekit/psrpc/blob/main/client.go#L198